### PR TITLE
Add new project property "TargetFrameworkMonikers" that lists all specified target monikers in the project.

### PR DESCRIPTION
**Pull request by jmarolf**
**Customer scenario**

.NET Core projects have added a new concept of being able to target multiple frameworks in a single project. We have asks from some MVPs to invest more in that feature and we need to make decisions on investment based on this data.

**Bugs this fixes:** 

https://github.com/dotnet/project-system/issues/1764

**Workarounds, if any**

None

**Risk**

Low.  This exposes a property that CPS will log telemetry on

**Performance impact**

Low. This exposes a property that CPS will log telemetry on

**Is this a regression from a previous update?**

No

### CODE REVIEW

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/InterceptedProjectProperties/TargetFrameworkMonikersValueProvider.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+        public override async Task<string> OnGetEvaluatedPropertyValueAsync(string evaluatedPropertyValue, IProjectProperties defaultProperties)
+        {
+            var configuration = await _properties.GetConfigurationGeneralPropertiesAsync().ConfigureAwait(true);
+            var currentTargetFramework = (string)await configuration.TargetFramework.GetValueAsync().ConfigureAwait(true);
+            var currentTargetFrameworks = (string)await configuration.TargetFrameworks.GetValueAsync().ConfigureAwait(true);
+            if (!string.IsNullOrEmpty(currentTargetFrameworks))
+            {
+                // sorting the target frameworks so projects that specify the same set of frameworks return the same string
+                return string.Join(";", ParseTargetFrameworks(currentTargetFrameworks).Sort());
```
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/InterceptedProjectProperties/TargetFrameworkMonikersValueProvider.cs]: Where is ParseTargetFrameworks defined? 

We're not supposed to be reading the TargetFrameork or TargetFrameworks properties - these are user defined properties with user defined values, I'm actively reducing our dependency on these properties within the project, they aren't supposed to be interpreted - that's the build's job.

Instead, we should be reading the TargetFrameworkMoniker property from all active configs which already has the parsed moniker or explicitly specified moniker.

Comment (jmarolf) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/InterceptedProjectProperties/TargetFrameworkMonikersValueProvider.cs]: ParseTargetFrameworks is defined in [TargetFrameworkProjectConfigurationDimensionProvider](https://github.com/dotnet/project-system/blob/master/src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Configuration/TargetFrameworkProjectConfigurationDimensionProvider.cs).  The assumption here was that we wanted to log the values the users are actually specifying even if they are invalid.  I don't have any problem with changing it, but how will the telemetry logs be better with the full TargetFrameworkMoniker string?

**Comment by srivatsn** [comment]
Can you sync up with  @adrivanvmsft on testing the change and once we have that, tag Matt to bring to shiproom? FYI @natidea

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/InterceptedProjectProperties/TargetFrameworkMonikersValueProvider.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+using static Microsoft.VisualStudio.ProjectSystem.Build.TargetFrameworkProjectConfigurationDimensionProvider;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Properties.InterceptedProjectProperties
+{
+    [ExportInterceptingPropertyValueProvider("TargetFrameworkMonikers", ExportInterceptingPropertyValueProviderFile.ProjectFile)]
```
Comment (srivatsn) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/InterceptedProjectProperties/TargetFrameworkMonikersValueProvider.cs]: If you're using ImplicitProjectfile, you'd have to export this differently I believe.

**Comment by jmarolf** [comment]
Got confirmation that this is working on the CPS side.  tagging @MattGertz for approval

**Comment by jmarolf** [comment]
NOTE: Telemetry side of this PR is [here](https://devdiv.visualstudio.com/DevDiv/_git/VS/pullrequest/64698)

**Comment by MattGertz** [comment]
@jmarolf I just want to confirm that this is for 15.2?  What's the VSO number for that?

**Comment by natidea** [comment]
I believe the bug is [383404](https://devdiv.visualstudio.com/DevDiv/_workitems?id=383404&_a=edit)

**Comment by MattGertz** [comment]
Thx.  Pending director approval, then Shiproom.

**Comment by natidea** [comment]
Approved to merge via [410926](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/410926)

**Pull request merged**